functions.inc.php

```diff
@@ -260,6 +260,33 @@ function check_domain($domain) {
     return '';
 }
 
+/**
+ * Checks if a domain is local
+ * @param string $domain
+ * @return string empty if the domain is valid, otherwise string with the errormessage
+ */
+function check_localaliasonly($domain) {
+    // If emailcheck_localaliasonly is set to 'YES', disallow aliases to remote servers (but allow aliases on this server)
+    if (Config::bool('emailcheck_localaliasonly')) {
+        // get the domain part of the e-mail
+        list(/*NULL*/, $domain) = explode('@', $domain);
+        
+        // get all domains managed on this system by postfixadmin
+        $domains = list_domains();
```

Comment:
Would it make sense to limit this to domains visible to the currently logged in admin? 

`list_domains_for_admin($username)` does that.

Background: By allowing all domains, this could leak information if a domain is hosted in the same PostfixAdmin instance (by trying various alias targets), even if an admin doesn't have permissions for this domain.

Sidenote: Since this function is only used by AliasHandler, I'd recommend to have it there instead of making functions.inc.php even bigger.

Reply:
I think that could easily be configurable even:
`$domains = ( (Config::bool(‘emailcheck_localaliasownedonly’)) ? list_domains_for_admin($username) : list_domains() );` 

My end goal was more to prevent forwards to hotmail/gmail (in favour of pulling such data instead of pushing it via SMTP), but your point that this could leak data is certainly valid.

I’d think that’s appropriate to use list_domains_for_admin(), so I’ll agree with you.

Reply:
>This could leak information if a domain is hosted in the same PostfixAdmin instance 

I don't think such leak is really critical, because these details could be easily explored from DNS.